src/vtbackend/Terminal.cpp

```diff
@@ -1139,6 +1139,44 @@ string Terminal::extractSelectionText() const
     }
 }
 
+string Terminal::extractSelectionTextAsHtml() const
+{
+    auto const _lock = scoped_lock { *this };
+
+    if (!selection_)
+        return "";
+
+    if (isPrimaryScreen())
+    {
+        auto se = SelectionRenderer<PrimaryScreenCell> { *this, pageSize().columns.as<ColumnOffset>() - 1 };
+        terminal::renderSelection(*selection_, [&](CellLocation _pos) { se(_pos, primaryScreen_.at(_pos)); });
+        return se.finish();
+    }
+    else
+    {
+        auto se = SelectionRenderer<AlternateScreenCell> { *this, pageSize().columns.as<ColumnOffset>() - 1 };
+        terminal::renderSelection(*selection_,
+                                  [&](CellLocation _pos) { se(_pos, alternateScreen_.at(_pos)); });
+        return se.finish();
+    }
+}
+
+string Terminal::makeCssTextFormatting(CellFlags cellFlags) const
+{
+    constexpr auto allFlags = 31; //0001 1111
+    string result {"#myDIV {}\n"};
+    switch(cellFlags)
+    {
+        case CellFlags::Bold: result.append("font-weight: bold;\n"); break;
+        case CellFlags::Italic: result.append("font-style: italic;\n"); break;
+        case CellFlags::Underline: result.append("text-decoration: underline;\n"); break;
+        case CellFlags::DottedUnderline: result.append("text-decoration: underline dotted;\n"); break;
+        default: 
+    }
+
+    return result+ "\n}\n";
```

Comment:
This function should not be part of the `Terminal` class. Please have a look at `VTWriter`, which basically translates the grid into the VT format as well.

You will need something similar to `VTWriter` but for HTML+CSS. You could give it the name `HtmlWriter` or `HtmlBuilder`.